openpgp/packet/config.go

```diff
@@ -105,6 +104,20 @@ func (c *Config) PasswordHashIterations() int {
 	return c.S2KCount
 }
 
+func (c *Config) RSAModulusBits() int {
+	if c == nil || c.RSABits == 0 {
+		return 2048
+	}
+	return c.RSABits
+}
+
+func (c *Config) PublicKeyAlgorithm() PublicKeyAlgorithm {
+	if c == nil || c.Algorithm == 0 {
+		return PubKeyAlgoRSA
+	}
+	return c.Algorithm
+}
+
```

Comment:
I don't like the idea of putting just part of the validation here, then part in the key generation, but I do agree that we should disable CAST5 and 3DES keys in a separate PR.

Reply:
Yeah I mean I agree, we should put that in this file when we do. 

> then part in the key generation

Are you referring to
https://github.com/ProtonMail/crypto/blob/facc4c5037e6eb2197b064537f6aecaa33d8088d/openpgp/key_generation.go#L163? I think it kinda makes sense to have it there because for example we don't support ECDSA key generation, but it's still a perfectly valid algorithm. So maybe the error message should be a bit more specific.
(On the other hand, I'm not sure in which other situation this function would be used, but maybe that's an argument for not having these functions at all.)